Allow for customization of pipeline templates (#2701)
kedro-org/kedro: docs/source/nodes_and_pipelines/modular_pipelines.md

If you want to do the reverse and remove a modular pipeline, you can use ``kedro pipeline delete <pipeline_name>`` to do so.

### Custom templates

If you want to generate a pipeline with a custom Cookiecutter template, you can save it in `<project_root>/templates/pipeline`.
The `kedro pipeline create` command will pick up the custom template in your project as the default. You can also specify the path to your custom
Cookiecutter pipeline template with the `--template` flag like this:
```bash
kedro pipeline create <pipeline_name> --template <path_to_template>
```

> A template folder passed through the command line using the `--template` argument will take precedence over any local templates.

Reviewer: ```suggestion
A template folder passed to `kedro pipeline create` using the `--template` argument will take precedence over any local templates.
```

> Kedro supports having a single pipeline template in your project. If you need to have many pipeline templates, consider saving them in a

Reviewer: ```suggestion
Kedro supports having a single pipeline template in your project. If you need to have multiple pipeline templates, consider saving them in a
```

separate folder and pointing to them with the `--template` flag.

#### Creating custom templates

> It is up to you to create functional Cookiecutter templates for custom modular pipelines. Please ensure you understand the

Reviewer: ```suggestion
It is your responsibility to create functional Cookiecutter templates for custom modular pipelines. Please ensure you understand the
```

basic structure of a modular pipeline. Your template should render to a valid, importable Python module containing a
`create_pipeline` function at the top level that returns a `Pipeline` object. You will also need appropriate

> `config` and `tests` subdirectories that will be copied to the project `config` and `tests` directories when creating a pipeline.

Reviewer: ```suggestion
`config` and `tests` subdirectories that will be copied to the project `config` and `tests` directories when the pipeline is created.
```

> Currently, the `config` and `tests` directories need to follow the exact same layout as in the default template and cannot

Reviewer: ```suggestion
Currently, the `config` and `tests` directories need to follow the same layout as in the default template and cannot
```

> be customized, though the contents of the parameters and actual test file can be changed. File and folder names or structure

Reviewer: ```suggestion
be customised, though the contents of the parameters and actual test file can be changed. File and folder names or structure
```

> do not matter beyond that and can be customized according to your needs. You can use [the

Reviewer: ```suggestion
do not matter beyond that and can be customised according to your needs. You can use [the
```

default template that Kedro](https://github.com/kedro-org/kedro/tree/main/kedro/templates/pipeline) uses as a starting point.

Pipeline templates are rendered using [Cookiecutter](https://cookiecutter.readthedocs.io/), and must also contain a `cookiecutter.json`
See the [`cookiecutter.json` file in the Kedro default template](https://github.com/kedro-org/kedro/tree/main/kedro/templates/pipeline/cookiecutter.json) for an example.
It is important to note that if you are embedding your custom pipeline template within a

> Kedro starter template, you must tell cookiecutter not to render this template when creating a new project from the starter. To do this,

Reviewer: "Cookiecutter" rather than "cookiecutter" ??
